addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -313,6 +313,75 @@ def test_mailing_computed_fields_default(self):
         })
         self.assertEqual(literal_eval(mailing.mailing_domain), [('email', 'ilike', 'test.example.com')])
 
+    @users('user_marketing')
+    def test_mailing_computed_fields_default_email_from(self):
+        # Testing if the email_from is correctly computed when an
+        # mail.alias.domain is set
+
+        # Case mail.alias.domain is set and no outgoing email server for mass_mailing
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing no default mail server for mass mailing',
+            'subject': 'Test',
+            'mailing_type': 'mail',
+            'body_html': '<p>Hello <t t-out="object.name"/></p>',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+        })
+        self.assertEqual(mailing.email_from, self.env.user.email_formatted)
```

Comment:
Heyho, not sure I am getting this point.
* You want me to have a use case where we test when the user does not have an email defined?
or
* Change the assert, i.e. what we compare it to?

Meanwhile: force pushed new changes and tried implement your other remarks

Reply:
Just adding an assert to make sure there is indeed an email (ensure the test data is consistent), because if there's no email it falls back on the trivial case of "is the email_from empty", which doesn't tell us whether the compute ran.

Unlikely to matter but that's the line of reasoning.

Reply:
Well, the aim of that test context was to ensure that if no `mail_server_id` is set (for example the corresponding settings for Mail Marketing is unticked), it will follow the first `if` of the _compute_email_from, i.e.
```
if not server:
                mailing.email_from = mailing.email_from or user_email
```

In other words, `user_email = self.env.user.email_formatted`. So this test does not really tackle the `create_uid` fix we had to do to ensure email_from is not False.
In itself I am not checking if the email_from is empty (it should always have some kinda of default), but more the expected output when based on the current _compute logic.

Also you say:
> user_marketing doesn't technically need to have an email
But unless mistaken, in a real case a user has always an email attached (either in the `res.users` record or the related `res.partner` record, for which `email` is a required field). 

Am I missing something?

Reply:
Mm, I don't think email is ever required. login on users is but it doesn't have to be their email (see our own instance).

But yes this works either way, just feels like it goes by an assumption that's not necessarily matched. As you say, it's not the goal of the test anyway though so :shrug:

Reply:
Yeah, but as far as I can see, their related `res.partner` record should have an email (even-though it is not necessarily the same as the login). So I guess in a normal flow, your should always have a user email. But there might be alternative path (internal, or direct SQL manipulations) I might be missing.